runtime/permissions/mod.rs

```diff
@@ -226,22 +228,267 @@ impl AsRef<str> for EnvVarName {
   }
 }
 
+pub trait Descriptor: Eq + Clone {
+  fn type_name() -> &'static str;
+  fn type_description() -> &'static str;
+  fn name(&self) -> Cow<str>;
+  fn stronger_than(&self, other: &Self) -> bool {
+    self == other
```

Comment:
This is a bit strange. Shouldn't it be `self > other`?

Reply:
> This is a bit strange. Shouldn't it be `self > other`?

This is the default implementation which specifies no _stronger than_ relationship. But since it's a non-strict _stronger than_, the only time it's true is when the descriptors are the same.